### PR TITLE
Support using SLF4J implementations for logging in the bundle

### PR DESCRIPTION
**Pull request by dagnir**
## Motivation and Context

The bundle shades the SLF4J classes. By adding a bridging mechanism, this allows logging through SLF4J to continue to work even when using the bundle.

For context, this will not change the user experience for bundle users; i.e. they will not need to pull in bundle-logging-bridge or any of the new modules. They will all be shaded into bundle.

To get around the shading, the SDK bundle is moved to `bundle-sdk`, so that we can include the `bundle-logging-bridge` artifacts in the final `bundle` artifact without the the SLF4J classes being shaded.
See the following classes for how SLF4J loads its bindings

- [LoggerFactory.java](https://github.com/qos-ch/slf4j/blob/0b97c416e42a184ff9728877b461c616187c58f7/slf4j-api/src/main/java/org/slf4j/LoggerFactory.java#L64)
- [MarkerFactory.java](https://github.com/qos-ch/slf4j/blob/0b97c416e42a184ff9728877b461c616187c58f7/slf4j-api/src/main/java/org/slf4j/MarkerFactory.java#L44)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: bundle-logging-bridge/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>annotations</artifactId>
+            <version>${awsjavasdk.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <!-- Provided scope as the 'thirdparty.org.slf4j' will be in the shaded bundle ill be in the bundle -->
```
Comment (millems) [inline bundle-logging-bridge/pom.xml:57]: Nit: extra words

File: bundle-logging-bridge/src/main/java/software/amazon/awssdk/thirdparty/org/slf4j/impl/internal/ILoggerFactoryAdapter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.thirdparty.org.slf4j.ILoggerFactory;
+import software.amazon.awssdk.thirdparty.org.slf4j.Logger;
+
+/**
+ * Adapts a normal, unshaded {@link org.slf4j.ILoggerFactory} to the shaded {@link ILoggerFactory}.
+ */
+@SdkInternalApi
+
```
Comment (millems) [inline bundle-logging-bridge/src/main/java/software/amazon/awssdk/thirdparty/org/slf4j/impl/internal/ILoggerFactoryAdapter.java:26]: nit: extra newline

**Review by millems — COMMENTED** [review]
Will we have to update these extra classes as slf4j evolves?

File: third-party/third-party-slf4j-api/pom.xml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>${slf4j.version}</version>
+            <optional>true</optional>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <!-- Shade Jackson -->
```
Comment (cenedhryn) [inline third-party/third-party-slf4j-api/pom.xml:55]: Nit: It's not Jackson here

**Comment by dagnir** [comment]
> Will we have to update these extra classes as slf4j evolves?

Possibly when we update `slf4j.version`, but SLF4J is compatible within minor versions, at least so not for any `1.7.x` updates.

**Comment by dagnir** [comment]
See the following classes for how SLF4J loads its bindings

- [LoggerFactory.java](https://github.com/qos-ch/slf4j/blob/0b97c416e42a184ff9728877b461c616187c58f7/slf4j-api/src/main/java/org/slf4j/LoggerFactory.java#L64)
- [MarkerFactory.java](https://github.com/qos-ch/slf4j/blob/0b97c416e42a184ff9728877b461c616187c58f7/slf4j-api/src/main/java/org/slf4j/MarkerFactory.java#L44)

**Comment by dagnir** [comment]
For context, this *will __not__* change the user experience for `bundle` users; i.e. they will not need to pull in `bundle-logging-bridge` or any of the new modules. They will all be shaded into `bundle`.

**Review by cenedhryn — APPROVED** [review]

**Pull request merged**